.github/workflows/gradle-check.yml

```diff
@@ -48,6 +51,8 @@ jobs:
           echo "pr_to_clone_url=$repo_url" >> $GITHUB_ENV
           echo "pr_title=Push trigger $branch_name $ref_id $repo_url" >> $GITHUB_ENV
           echo "pr_number=Null" >> $GITHUB_ENV
+          echo "pr_owner=Null" >> $GITHUB_ENV
+          echo "commit_details=$(jq --raw-output .head_commit.message $GITHUB_EVENT_PATH)" >> $GITHUB_ENV
```

Comment:
Might be worthwhile looking into the `sender` property on the payload [1], has details about the pull request that can be referenced.

- [1] https://docs.github.com/en/webhooks/webhook-events-and-payloads#push

Reply:
If you're talking about being able to identify the user who raised the PR that got merged, then sender is not the right place. The sender contains details of the user to merged the PR. 
However, there is a `commit` block that contains a list of all the commits that happened on the PR and if they are in the order in which they were pushed onto the PR then I believe the author of the first commit can easily be assumed to be the main author of the PR. Here is an example that I was working on:
```
"commits": [
    {
      "author": {
        "email": "160282956+noobie3007@users.noreply.github.com",
        "name": "noobie3007",
        "username": "noobie3007"
      },
      "committer": {
        "email": "noreply@github.com",
        "name": "GitHub",
        "username": "web-flow"
      },
      "distinct": false,
      "id": "0fcd01cbf6a2e8b69bd6b312ae9968b9cbf8b2cf",
      "message": "Update _config-dev.yml\n\nSigned-off-by: noobie3007 <160282956+noobie3007@users.noreply.github.com>",
      "timestamp": "2024-03-26T14:07:18-07:00",
      "tree_id": "7179ae117fdf223a9449d326aaaa8b451746aa9f",
      "url": "https://github.com/rishabh6788/project-website/commit/0fcd01cbf6a2e8b69bd6b312ae9968b9cbf8b2cf"
    },
    {
      "author": {
        "email": "rishabhksingh@gmail.com",
        "name": "Rishabh Singh",
        "username": "rishabh6788"
      },
      "committer": {
        "email": "noreply@github.com",
        "name": "GitHub",
        "username": "web-flow"
      },
      "distinct": true,
      "id": "2e1ec51a177d728acec646b019de898a3142a8f5",
      "message": "Merge pull request #2 from noobie3007/main\n\nUpdate _config-dev.yml",
      "timestamp": "2024-03-26T14:23:50-07:00",
      "tree_id": "5483ef5beff17db67c4de6899154ef85d39cfd64",
      "url": "https://github.com/rishabh6788/project-website/commit/2e1ec51a177d728acec646b019de898a3142a8f5"
    }
  ],
  ```
  
 Let me your thoughts.

Reply:
> `sender` object
> The GitHub user that triggered the event. This property is included in every webhook payload.

This seems like it would allow a traceback to the pull request, if the sender was triggered by a pull request, but I haven't used this in practice to know what it contains.